pkg/controllers/customeradmin/group_mapping_test.go

```diff
@@ -134,32 +159,50 @@ func TestFromMSGraphGroup(t *testing.T) {
 			},
 		},
 		{
-			name:          "guest member reconcile",
+			name:          "guest member reconcile external tree",
 			kubeGroupName: osaCustomerAdmins,
 			want1:         true,
 			kubeGroup: &v1.Group{
 				ObjectMeta: meta_v1.ObjectMeta{
 					Name: osaCustomerAdmins,
 				},
-				Users: []string{"foo@bar.com", "bar@foo.com"},
+				Users: []string{"live.com#user@homecorp.com", "user@nativecorp.com", "user@guestcorp.com"},
 			},
 			msGroupMembers: []graphrbac.User{
 				{
-					Mail:         to.StringPtr("foo@bar.com"),
-					UserType:     graphrbac.Guest,
-					MailNickname: to.StringPtr("#EXT#foo@bar.com"),
+					// ocpUser: live.com#user@homecorp.com
```

Comment:
Comments around all these usecases and 3 usecases defines 
```
1. owner mail = nil
2. guest (EXT) live.com#user@example.com
3. guest (ext) no-domain@example.com
4. Normal user - member UserPrincipalName
```

Reply:
put these comments in the main function and link to it here